config/muxcable.py

```diff
@@ -0,0 +1,175 @@
+            click.echo("Got invalid asic index for port {}, cant retreive mux status".format(port))
+            return
+
+        y_cable_asic_table = y_cable_tbl.get(asic_index, None)
+        if y_cable_asic_table is not None:
+            y_cable_asic_table_keys = y_cable_asic_table.getKeys()
+            if port in y_cable_asic_table_keys:
+                port_status_dict = {}
+                (status, fvs) = y_cable_asic_table.get(str(port))
+                if status is not True:
+                    click.echo("could not retrieve port values for port".format(port))
+                fvp = dict(fvs)
+                state_value = fvp.get("state",None)
+                if (state == "active" and state_value == "active") or (state == "auto" and state_value == "active"):
+                    # status is already active, so right back ok
+                    fvs = swsscommon.FieldValuePairs([('status', 'Ok')])
+                    y_cable_update_tbl[asic_index].set(logical_port_name, fvs)
+                    port_status_dict = {'status': 'Ok'}
+                elif state == "active" and state_value == "standby":
+                    # Change of status recived, right back inprogress
+                    fvs = swsscommon.FieldValuePairs([('status', 'inprogress')])
+                    y_cable_update_tbl[asic_index].set(logical_port_name, fvs)
+                    port_status_dict = {'status': 'inprogress'}
+                else:
+                    #Everything else to be treated as failure
+                    fvs = swsscommon.FieldValuePairs([('status', 'Failed')])
+                    y_cable_update_tbl[asic_index].set(logical_port_name, fvs)
+                    port_status_dict = {'status': 'Failed'}
+
+                if json_flag:
+                    click.echo("muxcable Ports status : \n {}".format(json.dumps(port_status_dict, indent=4)))
+                else:
+                    headers = ['port', 'status']
+                    data = sorted([(k,v) for k,v in port_status_dict.items()])
+                    click.echo(tabulate(data, headers=headers))
+
+            else:
+                click.echo("this is not a valid port present on mux_cable".format(port))
+        else:
+            click.echo("there is not a valid asic table for this asic_index".format(asic_index))
+
+
+    elif port == "all" and port is not None:
+
+        for namespace in namespaces:
+            asic_id = multi_asic.get_asic_index_from_namespace(namespace)
+            for logical_port in port_table_keys[asic_id]:
+                port_status_dict = {}
+                (status, fvs) = y_cable_asic_table.get(str(logical_port))
+                if status is not True:
+                    click.echo("could not retrieve port values for port".format(logical_port))
+                fvp = dict(fvs)
+                state_value = fvp.get("state",None)
+                if (state == "active" and state_value == "active") or (state == "auto" and state_value == "active"):
```

Comment:
The code here is common between all and single port. can it be refactored and called as separate method?

Reply:
refactored, fixed